File: 0720-longest-word-in-dictionary/0720-longest-word-in-dictionary.ts
function longestWord(words: string[]): string {
    words.sort();
    let longestWord = "";
    let builtWord = new Set();
    for(const word of words){
        if(builtWord.has(word.slice(0,-1)) || word.length===1){
           builtWord.add(word);
            if(word.length>longestWord.length) longestWord = word;
        }
    }
    return longestWord;
};

File: 0720-longest-word-in-dictionary/0720-longest-word-in-dictionary.py
class Solution:
    def longestWord(self, words: List[str]) -> str:
        words.sort()
        longest_word = ""
        built_word = set()
        for word in words:
            if len(word)==1 or (word[:-1] in built_word):
                built_word.add(word)
                if len(word) > len(longest_word):
                    longest_word = word 
        return longest_word